packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_base.dart

```diff
@@ -39,19 +42,21 @@ class AmplifyStorageS3Dart extends StoragePluginInterface<
     S3StorageRemoveOptions,
     S3StorageRemoveManyOperation,
     S3StorageRemoveManyOptions,
-    S3StorageItem> with AWSDebuggable, AWSLoggerMixin {
+    S3StorageItem,
+    S3TransferProgress> with AWSDebuggable, AWSLoggerMixin {
   /// {@macro amplify_storage_s3_dart.amplify_storage_s3_plugin_dart}
-  AmplifyStorageS3Dart({
+  AmplifyStorageS3DartBase({
```

Comment:
I would inject the downloader instead of having a base class. This gets really messy since you need to add `protected` everywhere and, especially with the generics.